### PR TITLE
Possible fix delete lost plugin main menu items

### PR DESCRIPTION
**Pull request by ghost**
https://mantis.ilias.de/view.php?id=25730

This fix has helped

The bug, that only items which are `isCustom` can be deleted (See https://github.com/ILIAS-eLearning/ILIAS/blob/b27519cd72ab7ef0c7295aff5372ab13e5359812/Services/MainMenu/classes/Administration/class.ilMMSubItemGUI.php#L239-L242)

But it will return a 'non custom facade', because the class name of the identifaction is not the custom provder class (`ilMMCustomProvider`), instead it's the plugin provider class (See https://github.com/ILIAS-eLearning/ILIAS/blob/b27519cd72ab7ef0c7295aff5372ab13e5359812/Services/MainMenu/classes/Items/class.ilMMItemRepository.php#L176-L180)

Strangely some lost plugin items can be deleted without this fix, other not

This PR does not fix the wrong redirected url (Delete top items will redirect to sub items page)
I think, this always so in the table action command. It' that correct so? (See https://github.com/ILIAS-eLearning/ILIAS/blob/b27519cd72ab7ef0c7295aff5372ab13e5359812/Services/MainMenu/classes/Administration/class.ilMMTopItemTableGUI.php#L104)

### CODE REVIEW

File: Services/MainMenu/classes/Items/class.ilMMItemRepository.php [hunk 1]
```diff
@@ -173,7 +174,7 @@ public function getItemFacade(IdentificationInterface $identification = null) :
         if ($identification === null || $identification instanceof NullIdentification || $identification instanceof NullPluginIdentification) {
             return new ilMMNullItemFacade($identification ? $identification : new NullIdentification(), $this->main_collector);
         }
-        if ($identification->getClassName() === ilMMCustomProvider::class) {
+        if ($identification->getClassName() === ilMMCustomProvider::class || $identification instanceof PluginIdentification) {
```
Comment (chfsx) [inline Services/MainMenu/classes/Items/class.ilMMItemRepository.php]: @sragfw technically speaking, plugin items are not custom icons, so the check for `PluginIdentification` would be wrong here. Probably better would be to do another `ItemFacade` for all items of type `Lost`. In this facade one would implement that they are deleteable (only the information about the item in `ilMMItemStorage` and `ilMMItemTranslationStorage` can be deleted). would you have time to implement such a facade?

**Review by chfsx — CHANGES_REQUESTED** [review]

File: Services/MainMenu/classes/Items/class.ilMMItemRepository.php (hunk 1 above)
Comment (ghost) [reply Services/MainMenu/classes/Items/class.ilMMItemRepository.php]: @chfsx I think, I found an other possible fix to check `isDeletable` and not `isCustom`

Comment (ghost) [reply Services/MainMenu/classes/Items/class.ilMMItemRepository.php]: @chfsx Sorry, but I have currently no time to implement a new facade for lost items

For instance I don't get based on `$identification`, if the item is `Lost` ...

**Comment by kergomard** [comment]
@chfsx : We stumbled upon this PR in our biweekly TB meeting. Could you please merge this PR or provide your objections?
Thank you in advance!

**Comment by chfsx** [comment]
Thanks @sragfw ! In 6 and trunk this should be already fixed in another way

**Pull request merged**